src/main/java/com/exasol/adapter/dialects/mysql/MySqlColumnMetadataReader.java

```diff
@@ -32,8 +35,34 @@ public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
             return DataType.createTimestamp(false);
         case Types.BINARY:
             return DataType.createUnsupported();
+        case Types.LONGVARCHAR:
+            return convertVarChar(jdbcTypeDescription);
         default:
             return super.mapJdbcType(jdbcTypeDescription);
         }
     }
+
+    private DataType convertVarChar(final JdbcTypeDescription jdbcTypeDescription) {
+        final int size = getSize(jdbcTypeDescription);
+        final int octetLength = jdbcTypeDescription.getByteSize();
+        final DataType colType;
+        final DataType.ExaCharset charset = (octetLength == size) ? DataType.ExaCharset.ASCII
+                : DataType.ExaCharset.UTF8;
+        if (size <= DataType.MAX_EXASOL_VARCHAR_SIZE) {
+            final int precision = size == 0 ? DataType.MAX_EXASOL_VARCHAR_SIZE : size;
```

Comment:
Use a "if - else if" here for better readability.

Reply:
done